### ISSUE TITLE
I need to create a stream image from IHtmlImageElement

### ISSUE THREAD

**Issue by walysonmax** [AngleSharp/AngleSharp#726]
I need to create a stream image from IHtmlImageElement.

In HtmlUnit, I'm using this:

```
HtmlImage image;
ImageReader reader = image.getImageReader();
```

So, how could I make this in AngleSharp?

**Comment by mausworks** [comment]
You can (for instance) download the image via it's URL, and use the response body.

```csharp
var client = new HttpClient();
var response = await client.GetAsync(image.Src);
var stream = await response.Content.ReadAsStreamAsync();
```

There are likely ways to do this via AngleSharp abstractions as well.

**Comment by walysonmax** [comment]
@mausworks 
I can not make another request because the image source changes

**Comment by FlorianRappl** [comment]
@walysonmax I do not understand - how does the image source change? If it changes what do you expect to get from AngleSharp (in form of a stream)?

There are options in AngleSharp that AngleSharp loads the resource for you. In this case, however, you also would not get the network stream, rather just the memory stream. To get the network stream you could hook on to the requester doing the HTTP connections.

HTH!

**Comment by mausworks** [comment]
@walysonmax 

Then how are you supposed to "read the image"? The image data is not located in the HTML document, it's another resource (hence `src`), you *have to* download the image via a separate request, that's how the web works.

![image](https://user-images.githubusercontent.com/8259221/48938271-07414280-ef11-11e8-946f-06ea3b5bfd88.png)

**Comment by walysonmax** [comment]
I resolved for the issue #379 .

Thanks.
